src/main/java/model/process/analysis/operations/comparisons/TimeBetween.java

```diff
@@ -1,11 +1,168 @@
+		}
+		if (!value.getClass().equals(DateValue.class)) {
+			builder.addColumn((DataTable) getInput(), new TimeValue(null, null,
+					null), "Difference time");
+		}
+
+		setInput(builder.build());
+
+		DataTable result = getVars((DataTable) getInput());
+
+		return result;
+
+	}
+
+	private DataTable getVars(DataTable table) {
+		
+		if (!value.getClass().equals(TimeValue.class)) {
+			table.getRow(0).setValue(table.getColumn("Difference date"),
+					new PeriodValue(0, 0, 0));
+		}
+		if (!value.getClass().equals(DateValue.class)) {
+			table.getRow(0).setValue(table.getColumn("Difference time"),
+					new TimeValue(0, 0, 0));
+		}
+
+		for (int i = 1; i < table.getRowCount(); i++) {
+			DataRow current = table.getRow(i);
+			DataRow previous = table.getRow(i - 1);
+
+			if (value.getClass().equals(DateTimeValue.class)) {
+				LocalDateTime cur = (LocalDateTime) current.getValue(
+						table.getColumn(dateName)).getValue();
+				curdate = cur.toLocalDate();
+				curtime = cur.toLocalTime();
+
+				LocalDateTime prev = (LocalDateTime) previous.getValue(
+						table.getColumn(dateName)).getValue();
+				predate = prev.toLocalDate();
+				pretime = prev.toLocalTime();
+			}
+			if (value.getClass().equals(DateValue.class)) {
+				curdate = (LocalDate) current.getValue(
+						table.getColumn(dateName)).getValue();
+				predate = (LocalDate) previous.getValue(
+						table.getColumn(dateName)).getValue();
+			}
+			if (value.getClass().equals(TimeValue.class)) {
+				curtime = (LocalTime) current.getValue(
+						table.getColumn(dateName)).getValue();
+				pretime = (LocalTime) previous.getValue(
+						table.getColumn(dateName)).getValue();
+			}
+			
+			table = calculate(table, current);
+			
+		}
+		return table;
+	}
+	
+	private DataTable calculate(DataTable table, DataRow current) {
+		boolean negative = false;
+
+		if (!value.getClass().equals(DateValue.class)) {
+			Duration diffTime = Duration.between(pretime, curtime);
+			int timeConvert = 60;
+			hour = (int) diffTime.toHours();
+			min = ((int) diffTime.toMinutes()) - (hour * timeConvert);
+			sec = ((int) diffTime.getSeconds()) - (min * timeConvert)
+					- (hour * timeConvert * timeConvert);
+			negative = checkNegative();
+			DataValue val = new TimeValue(hour, min, sec);
+			current.setValue(table.getColumn("Difference time"), val);
+		}
+
+		if (!value.getClass().equals(TimeValue.class)) {
+			Period diffDate = Period.between(predate, curdate);
+			if (negative) {
+				diffDate = diffDate.minusDays(1);
+			}
+			DataValue value = new PeriodValue(diffDate.getYears(),
+					diffDate.getMonths(), diffDate.getDays());
+			current.setValue(table.getColumn("Difference date"), value);
+		}
+		
+		return table;
+	}
+
+	private boolean checkNegative() {
 
+		if (hour < 0 || min < 0 || sec < 0) {
+			if (hour < 0) {
+				hour *= -1;
+			}
+			if (min < 0) {
+				min *= -1;
+			}
+			if (sec < 0) {
+				sec *= -1;
+			}
+			return true;
+		} else if ((hour < 0 || min < 0 || sec < 0)
```

Comment:
de 1e ifstatement is true als deze ook true zou zijn, dus dit wordt nooit bereikt. Verder ben ik ook niet helemaal zeker of deze methode doet wat hij zou moeten doen. 

Ik vermoed dat deze methode ervoor moet zorgen dat het correcte tijdverschil wordt getoond. Stel wanneer event 1 op dag 1 om 12 u  plaats vindt en dat event 2 op dag 3 om 11 u plaats vindt. Ik vermoed dat je wilt dat het resultaat van deze methode dan is, dat het verschil tussen de 2 events 1 dag en 23 u is. Maar nu zou dat 2 dagen en 1 u zijn.